{"name": "test_add_jewelry_to_cart_firefox", "status": "broken", "statusDetails": {"message": "tests.core.exceptions.AllureStepError: Произошла ошбика во время шага тестирования: select jewelry topic в браузере: firefox", "trace": "self = <tests.domain.use_cases.add_jewelry_to_cart.AddJewelryToCart object at 0x105e60100>\n\n    def execute(self):\n        driver = make_driver(self.browser)\n    \n        open_drop_down_list(browser=self.browser, driver=driver)\n    \n        with allure.step(self.SELECT_JEWELRY_TOPIC):\n            try:\n>               works = driver.find_element(\n                    By.CSS_SELECTOR,\n                    \"#left_container > div > ul:nth-child(20) > li:nth-child(5) > a\"\n                )\n\nuse_cases/add_jewelry_to_cart.py:36: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:741: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n../../../.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x105e61ae0>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"Unable to locate element: #left_container > d.../content/shared/webdriver/Errors.sys.mjs:509:5\\\\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: #left_container > div > ul:nth-child(20) > li:nth-child(5) > a; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nE       WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5\nE       NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:509:5\nE       dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n../../../.venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <src.tests.domain.composite.TestUseCases object at 0x105e4f550>\n\n    @allure.story(make_story(Story.FIFTH, Browser.CHROME))\n    @allure.severity(SeverityType.BLOCKER)\n    def test_add_jewelry_to_cart_firefox(self):\n        use_case = AddJewelryToCart(browser=Browser.FIREFOX)\n    \n>       use_case.execute()\n\ncomposite.py:22: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <tests.domain.use_cases.add_jewelry_to_cart.AddJewelryToCart object at 0x105e60100>\n\n    def execute(self):\n        driver = make_driver(self.browser)\n    \n        open_drop_down_list(browser=self.browser, driver=driver)\n    \n        with allure.step(self.SELECT_JEWELRY_TOPIC):\n            try:\n                works = driver.find_element(\n                    By.CSS_SELECTOR,\n                    \"#left_container > div > ul:nth-child(20) > li:nth-child(5) > a\"\n                )\n                works.click()\n            except Exception:\n                make_screenshot(browser=self.browser, driver=driver)\n>               raise AllureStepError(browser=self.browser, step=self.SELECT_JEWELRY_TOPIC)\nE               tests.core.exceptions.AllureStepError: Произошла ошбика во время шага тестирования: select jewelry topic в браузере: firefox\n\nuse_cases/add_jewelry_to_cart.py:43: AllureStepError"}, "steps": [{"name": "make driver", "status": "passed", "start": 1709478889862, "stop": 1709478894127}, {"name": "open drop down list", "status": "passed", "start": 1709478894127, "stop": 1709478894383}, {"name": "select jewelry topic", "status": "broken", "statusDetails": {"message": "tests.core.exceptions.AllureStepError: Произошла ошбика во время шага тестирования: select jewelry topic в браузере: firefox\n", "trace": "  File \"/Users/cuasar/Documents/GitHub/Catch/src/tests/domain/use_cases/add_jewelry_to_cart.py\", line 43, in execute\n    raise AllureStepError(browser=self.browser, step=self.SELECT_JEWELRY_TOPIC)\n"}, "steps": [{"name": "make screenshot", "status": "passed", "attachments": [{"name": "firefox error", "source": "ee68e9ba-6186-4eb6-bc73-4c3413adb824-attachment.png", "type": "image/png"}], "start": 1709478894385, "stop": 1709478894592}], "start": 1709478894383, "stop": 1709478895235}], "start": 1709478889862, "stop": 1709478895235, "uuid": "a88fa1d8-3992-498a-b4b1-c902b8e2867d", "historyId": "d2125ba8eb5296347033053182884e0c", "testCaseId": "d2125ba8eb5296347033053182884e0c", "fullName": "tests.domain.composite.TestUseCases#test_add_jewelry_to_cart_firefox", "labels": [{"name": "story", "value": "Перейти в “Ювелирное искусство”, добавить первое изделие в корзину, проверить, что выбранный товар находится в корзине, стоимость товара не изменилась. in chrome"}, {"name": "severity", "value": "blocker"}, {"name": "parentSuite", "value": "tests.domain"}, {"name": "suite", "value": "composite"}, {"name": "subSuite", "value": "TestUseCases"}, {"name": "host", "value": "Cuasars-MacBook-Air.local"}, {"name": "thread", "value": "30816-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.domain.composite"}]}